library/ssl_cookie.c

```diff
@@ -38,8 +38,8 @@
 #include <string.h>
 
 /*
- * If DTLS is in use, then at least one of SHA-1, SHA-256, SHA-384 is
- * available. Try SHA-256 first, 384 wastes resources
+ * If DTLS is in use, then at least one of SHA-1, SHA-224, SHA-384 is
```

Comment:
The new version of the comment is not correct: DTLS uses SHA-256, not SHA-224, so we can't infer availability of SHA-224 from the fact that DTLS is enabled (this entire module depends on DTLS).

Also, the old version of the comment was sub-optimal: since Mbed TLS 3.0 and the removal of DTLS 1.0, if DTLS is in use, we know that one of SHA-256 or SHA-384 is available.

I think we should change the logic here and quit using SHA-224: it made sense when we had the chain implication DTLS => (SHA-256 or SHA-384) => ((SHA-224 and SHA-256) or (SHA-384 and SHA-512)) but the later part is no longer true. Since the only purpose of using SHA-224 instead of 256 was to save 4 bytes, I'd say it's not worth it and we should just use SHA-256 or SHA-384.

Reply:
Just one question here: is SHA1 still supported? Or should I change the comment and code to support only SHA256 and SHA384?

Reply:
Yes, please change to only SHA-256 and SHA-384. We only support DTLS 1.2, which depends on one of those two, and no longer support DTLS 1.0 (if you're wondering, there's no DTLS 1.1), which was the last version that could use SHA-1 instead.